### PR TITLE
Fix tautological conditions

### PR DESCRIPTION
**Pull request by silverwind**
As discovered by https://github.com/go-gitea/gitea/pull/30729.

### CODE REVIEW

File: routers/private/hook_post_receive.go [hunk 1]
```diff
@@ -117,16 +117,14 @@ func HookPostReceive(ctx *gitea_context.PrivateContext) {
...
-					return
-				}
+			var err error
+			gitRepo, err = gitrepo.OpenRepository(ctx, repo)
+			if err != nil {
+				log.Error("Failed to open repository: %s/%s Error: %v", ownerName, repoName, err)
+				ctx.JSON(http.StatusInternalServerError, private.HookPostReceiveResult{
+					Err: fmt.Sprintf("Failed to open repository: %s/%s Error: %v", ownerName, repoName, err),
+				})
+				return
```
Comment (silverwind) [inline routers/private/hook_post_receive.go:127]: `gitRepo` is declared below, as far as I can tell, the variable is always `nil` at this point: https://github.com/go-gitea/gitea/blob/52c003acd790449866f0acfc131198c9094f6838/routers/private/hook_post_receive.go#L42-L46

File: services/convert/issue.go [hunk 2]
```diff
@@ -211,13 +211,11 @@ func ToLabel(label *issues_model.Label, repo *repo_model.Repository, org *user_m
 		IsArchived:  label.IsArchived(),
 	}
 
+	labelBelongsToRepo := label.BelongsToRepo()
+
 	// calculate URL
-	if label.BelongsToRepo() && repo != nil {
-		if repo != nil {
```
Comment (silverwind) [inline services/convert/issue.go:216]: This line was erroring. I removed this `if` and moved the logging call outside the condition so it's preserved.

File: services/auth/source/oauth2/providers.go [hunk 3]
```diff
@@ -182,7 +182,7 @@ func createProvider(providerName string, source *Source) (goth.Provider, error)
 	}
 
 	// always set the name if provider is created so we can support multiple setups of 1 provider
-	if err == nil && provider != nil {
+	if provider != nil {
```
Comment (silverwind) [inline services/auth/source/oauth2/providers.go:185]: `err` is guaranteed to be `nil` here.

**Comment by harryzcy** [comment]
There's another one at `modules/indexer/code/indexer.go:181` that `err` is always `nil`

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
> There's another one at `modules/indexer/code/indexer.go:181` that `err` is always `nil`

Fixed in https://github.com/go-gitea/gitea/pull/30735/commits/392a25b1e89684017ff6b9f7ea81d0a85b73c0b8 by removing the impossible to reach block.

**Comment by silverwind** [comment]
Two more issues fixed in `git_test.go`, I think this is ready for review/merge and I will handle the remaining errors in https://github.com/go-gitea/gitea/pull/30729.

**Review by lunny — APPROVED** [review]

File: services/convert/issue.go [hunk 4]
```diff
@@ -226,6 +224,10 @@ func ToLabel(label *issues_model.Label, repo *repo_model.Repository, org *user_m
 		}
 	}
 
+	if labelBelongsToRepo && repo == nil {
+		log.Error("ToLabel did not get repo to calculate url for label with id '%d'", label.ID)
+	}
```
Comment (wxiaoguang) [inline services/convert/issue.go:229]: I still have objection for doing pure-refactoring in this period.

And ideally, it should be like this:

```
if labelBelongsToRepo {
    if repo != nil {
        ...
    } else {
       log.Error
    }
} else {
    if org != nil {
        ...
    } else {
        log.Error(...)
    }
}
```

Comment (silverwind) [reply services/convert/issue.go:229]: Those sections are unlikely to be targeted by backports, so I think it's okay. Feel free to raise a PR for that.

**Pull request merged**